dev/com.ibm.ws.concurrent.persistent_fat_multiple/fat/src/com/ibm/ws/concurrent/persistent/fat/multiple/DatabaseContainerFactory.java

```diff
@@ -0,0 +1,144 @@
+ * The test container returned will be based on the 'fat.bucket.db.type' 
+ * system property. 
+ * 
+ * The 'fat.bucket.db.type' property is set to different databases
+ * by our test infrastructure when a fat-suite has the property
+ * 'fat.test.databases' set to true
+ * 
+ * TODO: Since this factory class is general can it be moved to fattest simplicity?
+ *
+ */
+public class DatabaseContainerFactory {
+	private static String database = System.getProperty("fat.bucket.db.type", "Derby");
+	private static GenericContainer<?> container = initContainer();
+	
+	@SuppressWarnings("rawtypes")
+	private static Class[] classes;
+	
+	/**
+	 * Reads the 'fat.bucket.db.type' system property and returns a
+	 * container based on that property.
+	 * 
+	 * If we are using a network database, a test container running
+	 * that database will be returned.  
+	 * [Postgre, DB2, Oracle, SQLServer, MySQL]
+	 * 
+	 * If we are using derby embedded or unsupported database
+	 * a no-op container will be returned.
+	 * [Derby, Informix, Sybase]
+	 * 
+	 * @param testClasses - Provide the name(s) of your test class(es). These will be passed to the test container which will then provide log output to the class(es) log.  
+	 * @return GenericContainer - Test test container.
+	 */
+	@SuppressWarnings("rawtypes")
+	public static GenericContainer<?> get(Class... testClasses) {
+		classes = testClasses;
+		return container;
+	}
+	
+	/**
+	 * Returns true if the test container is configured
+	 * for a network database.
+	 * This will need to be checked prior to setting
+	 * server environment variables for database connectivity
+	 * @return
+	 */
+	public static boolean isNetworkDatabase() {
+		return container instanceof  JdbcDatabaseContainer;
+	}
+	
+	/**
+	 * Returns the name of the database that will be 
+	 * used to generate a test container. 
+	 * @return String - name of database
+	 */
+	public static String usingDB() {
+		return database;
+	}
+	
+	/**
+	 * Private Method: used to initialize test container.
+	 */
+	private static GenericContainer<?> initContainer() {		
+    	switch(database) {
+    	case "Postgre":
+    		return new PostgreSQLContainer<>()
+    				.withLogConsumer(DatabaseContainerFactory::log);
+		case "DB2":
+			return new Db2Container()
+                    .acceptLicense()
+                    // Use 5m timeout for local runs, 15m timeout for remote runs
+                    .withStartupTimeout(Duration.ofMinutes(FATRunner.FAT_TEST_LOCALRUN ? 5 : 15))
+                    .withLogConsumer(DatabaseContainerFactory::log);
+		case "Oracle":
+			//TODO does IBM/Websphere have an oracle docker container that could be used here instead?
+			return new OracleContainer("oracleinanutshell/oracle-xe-11g")
+					.withLogConsumer(DatabaseContainerFactory::log);
+		case "SQLServer":
+			return new MSSQLServerContainer<>()
+					.withLogConsumer(DatabaseContainerFactory::log);
+		case "Derby":
+		default:
+			return new MockServerContainer();
```

Comment:
why do we need to use MockServer here? As far as I can tell it's not used aside from having _some_ container to return in the case of Derby.

Instead, lets implement a "dummy" `DerbyContainer` class that extends `JdbcDatabaseContainer` (see `CustomPostgreSQLContainer` for an example) and we can just override/hardcode the necessary methods to so things like `container.start()` simply no-op for the Derby case